x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/slm/SnapshotRetentionTask.java

```diff
@@ -240,8 +240,10 @@ void getAllRetainableSnapshots(Collection<String> repositories, ActionListener<M
             return;
         }
 
+        // don't time out on this request to not produce failed SLM runs in case of a temporarily slow master node
```

Comment:
nit: could this go on the line above the `setMasterNodeTimeout` method call?

Reply:
++